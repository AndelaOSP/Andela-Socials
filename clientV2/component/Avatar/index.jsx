import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description - This component displays a single Avatar
 *
 * @export Avatar
 * @param {object} { imgUrl, classes, altText }
 * @returns {JSX}
 */
export const Avatar = ({ imgUrl, classes, altText }) => {
  return <img className={`avatar ${classes}`} src={imgUrl} alt={altText} />;
}

Avatar.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  classes: PropTypes.string,
  altText: PropTypes.string
};

Avatar.defaultProps = {
  classes: '',
  altText: ''
};
