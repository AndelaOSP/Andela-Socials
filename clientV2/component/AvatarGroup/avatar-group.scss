@import '../../assets/style.scss';

.avatar-group {
  display: flex;
  align-items: center;
  z-index: 1;

  .avatar-item {
    width: 25px;
  }

  .img-count {
    display: inline-block;
    vertical-align: middle;
    margin-left: 25px;
    font-size: 1.125rem;
    color: map-get($colors, grey);
  }
}
