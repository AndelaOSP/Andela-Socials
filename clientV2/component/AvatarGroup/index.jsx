/* eslint-disable array-callback-return */
import React from 'react';
import PropTypes from 'prop-types';

import Avatar from '../Avatar';
import './avatar-group.scss';

/**
 * @description - This component displays group of Avatar
 *
 * @param {object} { classes, imgList }
 * @return {JSX}
 */
const AvatarGroup = ({ classes, imgList }) => (
  <div className="avatar-group">
    {imgList.map((img, index) => {
      if (index < 7) {
        return (
          <div className={`avatar-item ${classes}`}>
            <Avatar imgUrl={img.imgUrl} />
          </div>
        );
      }
    })}
    {imgList.length > 7 && <span className="img-count">{`+${imgList.length - 7}`}</span>}
  </div>
);

AvatarGroup.propTypes = {
  classes: PropTypes.string,
  imgList: PropTypes.arrayOf(PropTypes.shape).isRequired
};

AvatarGroup.defaultProps = { classes: '' };

export default AvatarGroup;
