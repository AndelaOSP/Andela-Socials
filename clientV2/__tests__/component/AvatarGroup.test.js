import React from 'react';
import { shallow } from 'enzyme';
import { AvatarGroup } from '../../component/AvatarGroup';
import { Avatar } from '../../component/Avatar';

describe('AvatarGroup Component', () => {
  let wrapper;
  const imgList = () => {
    const imgArr = [];
    for (let i = 0; i < 10; i++) {
      imgArr.push({ imgUrl: `//http:/img.jpg${i + 1}` });
    }
    return imgArr;
  };

  beforeEach(() => {
    const props = {
      imgList: imgList()
    };
    wrapper = shallow(<AvatarGroup {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display Avatar Component', () => {
    const avatar = wrapper.find(Avatar);
    expect(avatar.length).toBe(7);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display Avatar count if Avatar length is more than 8', () => {
    const avatarCount = wrapper.find('.img-count');
    expect(avatarCount.text()).toContain(3);
  });

  it('should not display Avatar count if Avatar length is less than 8', () => {
    const props = {
      imgList: [{ imgUrl: '//http:/img.jpg1' }]
    };
    const wrapper = shallow(<AvatarGroup {...props} />);
    const avatarCount = wrapper.find('.img-count');
    expect(avatarCount.length).toBe(0);
  });
});
