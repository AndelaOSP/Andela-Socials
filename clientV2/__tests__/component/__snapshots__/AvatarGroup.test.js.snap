// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup Component should display Avatar Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes=""
    imgList={
      Array [
        Object {
          "imgUrl": "//http:/img.jpg1",
        },
        Object {
          "imgUrl": "//http:/img.jpg2",
        },
        Object {
          "imgUrl": "//http:/img.jpg3",
        },
        Object {
          "imgUrl": "//http:/img.jpg4",
        },
        Object {
          "imgUrl": "//http:/img.jpg5",
        },
        Object {
          "imgUrl": "//http:/img.jpg6",
        },
        Object {
          "imgUrl": "//http:/img.jpg7",
        },
        Object {
          "imgUrl": "//http:/img.jpg8",
        },
        Object {
          "imgUrl": "//http:/img.jpg9",
        },
        Object {
          "imgUrl": "//http:/img.jpg10",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg1"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg2"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg3"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg4"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg5"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg6"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg7"
            />
          </div>,
          undefined,
          undefined,
          undefined,
        ],
        <span
          className="img-count"
        >
          +3
        </span>,
      ],
      "className": "avatar-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "img0",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg1"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img1",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg2"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img2",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg3"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img3",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg4"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img4",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg5"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg5",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img5",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg6"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg6",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img6",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg7"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg7",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      undefined,
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+3",
          "className": "img-count",
        },
        "ref": null,
        "rendered": "+3",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg1"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg2"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg3"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg4"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg5"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg6"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg7"
              />
            </div>,
            undefined,
            undefined,
            undefined,
          ],
          <span
            className="img-count"
          >
            +3
          </span>,
        ],
        "className": "avatar-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "img0",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg1"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img1",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg2"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img2",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg3"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img3",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg4"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img4",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg5"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg5",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img5",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg6"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg6",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img6",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg7"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg7",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        undefined,
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+3",
            "className": "img-count",
          },
          "ref": null,
          "rendered": "+3",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`AvatarGroup Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes=""
    imgList={
      Array [
        Object {
          "imgUrl": "//http:/img.jpg1",
        },
        Object {
          "imgUrl": "//http:/img.jpg2",
        },
        Object {
          "imgUrl": "//http:/img.jpg3",
        },
        Object {
          "imgUrl": "//http:/img.jpg4",
        },
        Object {
          "imgUrl": "//http:/img.jpg5",
        },
        Object {
          "imgUrl": "//http:/img.jpg6",
        },
        Object {
          "imgUrl": "//http:/img.jpg7",
        },
        Object {
          "imgUrl": "//http:/img.jpg8",
        },
        Object {
          "imgUrl": "//http:/img.jpg9",
        },
        Object {
          "imgUrl": "//http:/img.jpg10",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg1"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg2"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg3"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg4"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg5"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg6"
            />
          </div>,
          <div
            className="avatar-item "
          >
            <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg7"
            />
          </div>,
          undefined,
          undefined,
          undefined,
        ],
        <span
          className="img-count"
        >
          +3
        </span>,
      ],
      "className": "avatar-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "img0",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg1"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img1",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg2"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img2",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg3"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img3",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg4"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img4",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg5"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg5",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img5",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg6"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg6",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "img6",
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            altText=""
            classes=""
            imgUrl="//http:/img.jpg7"
          />,
          "className": "avatar-item ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "",
            "classes": "",
            "imgUrl": "//http:/img.jpg7",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      undefined,
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+3",
          "className": "img-count",
        },
        "ref": null,
        "rendered": "+3",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg1"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg2"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg3"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg4"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg5"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg6"
              />
            </div>,
            <div
              className="avatar-item "
            >
              <Unknown
                altText=""
                classes=""
                imgUrl="//http:/img.jpg7"
              />
            </div>,
            undefined,
            undefined,
            undefined,
          ],
          <span
            className="img-count"
          >
            +3
          </span>,
        ],
        "className": "avatar-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "img0",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg1"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img1",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg2"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img2",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg3"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img3",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg4"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img4",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg5"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg5",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img5",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg6"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg6",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "img6",
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              altText=""
              classes=""
              imgUrl="//http:/img.jpg7"
            />,
            "className": "avatar-item ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "",
              "classes": "",
              "imgUrl": "//http:/img.jpg7",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        undefined,
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+3",
            "className": "img-count",
          },
          "ref": null,
          "rendered": "+3",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
